_posts/2021-05-04-stability.md

```diff
@@ -0,0 +1,106 @@
+This can cause friction for downstream users today,
+but also means more-than-incremental changes for the future.
+We've optimized a little bit for long-term growth over short-term stability.
+
+
+### Motivation for change
+
+There are two structural things driving some of these changes:
+
+1.  An increase in computational scale
+2.  An increase in organizational scale
+
+
+### Computational Scale
+
+Dask today is used across a wider range of problems,
+a more diverse set of hardware,
+and at larger scales more routinely than before.
+
+Addressing this increase in scale across many dimensions has caused us to
+redesign Dask's internal infrastructure in several ways.
+
+1.  We've changed how Dask graphs are represented and communicated to the scheduler
+2.  We've pulled out Dask's internal state machines and made them more formalized
+3.  We've rewritten large chunks of the scheduler in Cython
+4.  We've overhauled how we serialize messages that go between all Dask servers
+5.  We're now tracking memory with much finer granularity than we did before
+6.  ... and more
+
+We've been doing all of these internal changes with minimal impact to the
+myriad of downstream user communities (Xarray, Prefect, RAPIDS, XGBoost, ...).
+This is largely due to those downstream developer communities,
+who help to identify, isolate, and work through the subtle tremors that occur
+on the surface when we make these subsurface shifts.
+
+
+### Organizational scale
+
+Historically Dask's core was maintained by a relatively small set of people,
+mostly at Anaconda.
+There were dozens of developers that worked on various `dask-foo` projects, but
+only a small group that thought about things like serialization, state
+machines, and so on.
+In particular I personally tracked every issue and knew the entire project.
+Whenever a potential conflict arose I was usually able to identify it early.
+
+This has all changed dramatically.
+
+First, there are now several multi-company teams working on different parts of
+Dask internals.
+
+Second, we've also taken some time to redesign parts of Dask internals to make them more maintainable.
+Dask scheduling is like a finely made clock.
+Historically parts of that clock were built and designed by individuals with a craftsman-like approach.
+Now we're redesigning things with more of a group mindset.
+This results in more maintainable designs,
+but it also means that we're taking apart the clock and putting it back together.
+It takes a little while to find all of the missing parts :)
+
+
+### How this affects you today
+
+This all started around when we switched to Calendar Versioning at the end of last year
+(Dask version `2.30.1` rolled over into `2020.12.0` last December).  You may
+have noticed
+
+1.  an increased sensitivity to version mismatches (as we change the Dask
+    protocol different versions of Dask can no longer talk to each other well)
+2.  releases with stability issues (2020.12 was particularly rough)
+3.  ...
```

Comment:
I cannot think of anything very concrete at the moment other than saying that we haven't recovered from `2020.12`, yet.